### DISCUSSION TITLE
osx pass qute-pass strange behaviour

### DISCUSSION THREAD

**Discussion by aleksejkrueger** [qutebrowser/qutebrowser#7528] [Q&A] [answered]
Hello,

first of all I would like to say thank you for this great browser that I use every day.

I have a strange problem.

**Version info**:

```
qutebrowser v2.4.0
Git commit: 36ffff2f6 on HEAD (2021-10-21 18:50:06 +0200)
Backend: QtWebEngine 5.15.2, based on Chromium 83.0.4103.122
Qt: 5.15.2
```

**Does the bug happen if you start with `--temp-basedir`?**: `yes`

**Description**

`OS: macOS 12.6.2 21G320 x86_64`

I use the password manager pass to manage my passwords.
I also use qute-pass to retrieve the passwords in qutebrowser.

But now there is one case where it works to retrieve the passwords and one case where it does not work.

I have installed qutebrowser twice.
Once as a virtual environment and once "https://github.com/qutebrowser/qutebrowser/releases/download/v2.4.0/qutebrowser-2.4.0.dmg" file downloaded and installed.
For some reason the homebrew formula does not work for me (but that's another problem).

I describe the case where it does **not** work:

It makes no difference whether I open Qutebrowser as an app or in the virtual environment and try to retrieve the password. I get the error in both cases 

```
Traceback (most recent call last):
  File "/Users/user/.qutebrowser/userscripts/qute-pass", line 63, in <module>
    import tldextract
ModuleNotFoundError: No module named 'tldextract'
```

Now the case where it works:

I use [url-view](https://packages.debian.org/sid/misc/urlview) from the [homebrew formula](https://formulae.brew.sh/formula/urlview) with the tmux-plugin: [tmux-urlview](https://github.com/tmux-plugins/tmux-urlview) to open a link from the terminal. Now when I try to invoke the password, everything works fine.

Shouldn't it work fine in both cases?

Is there a path missing in or something similar?

If anyone has the crucial clue for me on how to get it to work, I would be very grateful.

**Comment by The-Compiler** [comment]
Converting this to a discussion, as issues are for bug reports and feature requests - this doesn't seem like a bug.

**Comment by toofar** [comment, marked as answer]
There is a related issue about the path that userscripts see being different when run from a qutebrowser that is launched as an app (less stuff in path) vs a terminal: https://github.com/qutebrowser/qutebrowser/issues/4273

But when you open a link from tmux it should go to an existing qutebrowser instance. Maybe it's not finding the IPC socket for the existing instance and always opening a new one?

I would examine the environments of the three setups: qute 1) launched from the virtual env 2) launched from as an app 3) the one that magically makes things work. And try to see if you can spot anything different in the environment variables with paths in them.

One way to look at the environment is to run `:spawn /usr/bin/env` (or whatever the path to `env` is) and then look at the output in `:process`. You can compare the stuff listed in `:version` too.

**Comment by aleksejkrueger** [comment]
Thanks for your help.
It was a missing Path.

path from app version 

```
PATH=/usr/bin:/bin:/usr/sbin:/sbin
```

path from opend link with tmux

```
PATH=/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands
```

How should I apply the Path from the tmux opend qutebrowser to the normal app?

**Reply by toofar** [reply]
You can tell launchd to set a different path for _all_ applications: https://github.com/qutebrowser/qutebrowser/issues/4273#issuecomment-595892960

Or, since this is just for launching a child process you could change `os.environ["PATH"]` in your config.py.

**Comment by aleksejkrueger** [comment]
Ok the following solution works for me:


I have added this 

```
"""Qutebrowser configuration."""
import platform
import os
from os.path import expanduser

system = platform.system()
home_dir = expanduser('~')

if system == 'Darwin':

    path = os.pathsep + '/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands'
    os.environ['PATH'] = path
```

at the top of my `config.py`

and now the the userscript is able to retrieve the password from the repository.


Thank you very much for your help!
